generate.go

```diff
@@ -1,4 +1,38 @@
-//go:generate go-bindata -pkg=data -ignore=.*\.go -o data.go . ./benkeen/d3pie/ ./jiahuang/d3-timeline/ ./krisk/fuse/ ./zeroclipboard/zeroclipboard/ ./seiyria/bootstrap-slider/
-//go:generate go fmt .
+// +build generate
 
-package data
+package main
+
+import (
+	"log"
+	"net/http"
+	"os"
+	pathpkg "path"
+
+	"github.com/shurcooL/go/vfs/httpfs/filter"
+	"github.com/shurcooL/vfsgen"
+)
+
+func main() {
+	ignore := func(fi os.FileInfo, path string) bool {
+		switch {
+		case len(fi.Name()) >= 2 && fi.Name()[0] == '.': // Skip .files/.folders (but not ".").
+			return true
+		case pathpkg.Ext(fi.Name()) == ".go":
+			return true
+		case path == "/README.md":
+			return true
+		default:
+			return false
+		}
+	}
+	var data http.FileSystem = filter.NewIgnore(http.Dir("."), ignore)
+
+	err := vfsgen.Generate(data, vfsgen.Options{
+		Filename:     "data.go",
+		PackageName:  "data",
+		VariableName: "Assets",
+	})
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
```

Comment:
> One bothersome thought is how verbose this is. We've gone from a few `go:generate` lines to an entire script here. I think we should focus on making this minimal so that we're not regressing.

I understand the concern. I'll have to be upfront in saying that it's likely we'll have to regress in verbosity to some degree (perhaps reduced from current state) due to the nature of `vfsgen` being used as a Go library rather than a binary.

Executing a binary with command line flags will be much shorter (in number of lines, although the lines themselves can be pretty long).

If there's a way to execute Go code in a more succinct manner than relying on `go run filename.go`, for example by using [`goe`](https://github.com/shurcooL/goe#goe) tool, then it can be made much less verbose:

```Go
//go:generate goe `vfsgen.Generate(http.Dir("assets"), vfsgen.Options{Filename: "data.go", PackageName: "data", VariableName: "Assets"})`
```

As an example.

Even without using `goe`, you can shorten number of lines of code by having longer lines, e.g.:

```Go
vfsgen.Options{Filename: "data.go", PackageName: "data", VariableName: "Assets"}
```

Instead of the current version.

However, I don't think that succinctness would be beneficial. This is code that has to be written few times, but read and understood and perhaps modified a lot more often. I'd argue the current version should be much easier to read and make changes to, even non-trivial custom changes that `go-bindata` does not have a flag for right now. This additional flexibility comes at the cost of a 38 line `generate.go` script.

I'm not against reducing the verbosity, I just wanted to point out that trying to match the previous line count may not be an optimal goal.

To address the 2 specific comments you brought up:

> #### `Filename`, `PackageName`, and `VariableName`
>
> Let's make these the defaults that vfsgen provides us with, so we can omit these options entirely.

vfsgen can be used in at least two ways: to embed assets inside Go commands (binaries), or to embed assets inside Go libraries (non-main packages).

I have thought about it, and I think the command case is more common use case. It stands to reason to have the defaults optimized for that case, as they currently are. That's why I don't think this change can be made. Given that `appdash-data` is a more advanced use, it seems normal to have it supply custom options.

It's possible to get rid of the `Filename` option. I've included it to simply match current behavior, but if omitted, it will generate output in a standard `Assets_vfsdata.go` location.

> #### The ignore function...
> 
> ...is rather verbose too! I wonder if we couldn't add a glob-based helper function to filter or some other package?

Sure, a helper like that can absolutely be done, and I think it would fit well inside the `filter` package as you've shown.

One thing to be careful about, glob patterns, as described [here](https://godoc.org/path/filepath#Match), have some limitations. One of them is the inability to match a varied sequence of Separator characters. This may not be an issue, but I wanted to point it out.